File: backend/WebApi.Model/Insignia.cs
namespace WebApi.Model;

public class InsigniaInfo
{
    public int InsigniaId { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string? Descripcion { get; set; }
    public string? ImagenUrl { get; set; }
    public DateTime FechaObtenida { get; set; }
}

File: mobile/lib/models/insignia.dart
class Insignia {
  final int insigniaId;
  final String nombre;
  final String? descripcion;
  final String? imagenUrl;
  final DateTime fechaObtenida;

  Insignia({
    required this.insigniaId,
    required this.nombre,
    this.descripcion,
    this.imagenUrl,
    required this.fechaObtenida,
  });

  factory Insignia.fromJson(Map<String, dynamic> json) {
    return Insignia(
      insigniaId: json['insigniaId'],
      nombre: json['nombre'],
      descripcion: json['descripcion'],
      imagenUrl: json['imagenUrl'],
      fechaObtenida: DateTime.parse(json['fechaObtenida']),
    );
  }
}